### ISSUE TITLE
Add Random Forest mode

### ISSUE THREAD

**Issue by ekerazha** [catboost/catboost#545]
Both XGBoost and LightGBM support a "Random Forest" mode.

In XGBoost you set the  `num_parallel_tree` parameter with  `nrounds=1` (https://xgboost.readthedocs.io/en/latest/R-package/discoverYourData.html#special-note-what-about-random-forests).

In LightGBM you use the  `boosting='rf'` parameter (https://lightgbm.readthedocs.io/en/latest/Parameters.html?highlight=Random%20Forest).

Would it be possible to add a "Random Forest" mode to CatBoost?

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
Hi ekerazha!
Yes, we are ready to review a pull request that adds random forests to CatBoost.
In our experience, however, random forests achieve lower quality than gradient boosting.
